x/ratelimit/types/message_add_rate_limit.go

```diff
@@ -0,0 +1,74 @@
+package types
+
+import (
+	"strings"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+// Msg type for MsgAddRateLimit
+const TypeMsgAddRateLimit = "add_rate_limit"
+
+var _ sdk.Msg = &MsgAddRateLimit{}
+
+func NewMsgAddRateLimit(creator string, denom string, channelId string, maxPercentSend uint64, maxPercentRecv uint64, durationHours uint64) *MsgAddRateLimit {
+	return &MsgAddRateLimit{
+		Creator:        creator,
+		Denom:          denom,
+		ChannelId:      channelId,
+		MaxPercentSend: maxPercentSend,
+		MaxPercentRecv: maxPercentRecv,
+		DurationHours:  durationHours,
+	}
+}
+
+func (msg *MsgAddRateLimit) Route() string {
+	return RouterKey
+}
+
+func (msg *MsgAddRateLimit) Type() string {
+	return TypeMsgAddRateLimit
+}
+
+func (msg *MsgAddRateLimit) GetSigners() []sdk.AccAddress {
+	creator, err := sdk.AccAddressFromBech32(msg.Creator)
+	if err != nil {
+		panic(err)
+	}
+	return []sdk.AccAddress{creator}
+}
+
+func (msg *MsgAddRateLimit) GetSignBytes() []byte {
+	bz := ModuleCdc.MustMarshalJSON(msg)
+	return sdk.MustSortJSON(bz)
+}
+
+func (msg *MsgAddRateLimit) ValidateBasic() error {
+	_, err := sdk.AccAddressFromBech32(msg.Creator)
+	if err != nil {
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "invalid creator address (%s)", err)
+	}
+
+	if msg.Denom == "" {
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "invalid denom")
+	}
+
+	if !strings.HasPrefix(msg.ChannelId, "channel-") {
```

Comment:
I think we should check one more case that channelId is exactly _channel-_.
What do you think about checking length of channel id too? 
For example, we can accept channel ids that length is bigger than 8 so that it should always look like this `channel-xxx`

Reply:
Agreed, I'll add a regex check